content/en/blog/_posts/2021-08-20-seccomp-default.md

```diff
@@ -0,0 +1,257 @@
+
+- _Recommended_: Analyzing the code (manually or by running the application with
+  [strace][strace]) for any executed syscalls which may be blocked by the
+  default profiles. If that's the case, then you can create a custom seccomp
+  profile based on the default by adding the additional syscalls to the
+  `"action": "SCMP_ACT_ALLOW"` section.
+
+- _Recommended_: Run the application against an end-to-end test suite to trigger
+  all relevant code paths with `RuntimeDefault` enabled. If a test fails, use
+  the same mitigation as mentioned above.
+
+- _Optional_: Create a custom seccomp profile based on the default and change
+  its default action from `SCMP_ACT_ERRNO` to `SCMP_ACT_LOG`. This means that
+  the seccomp filter for unknown syscalls will have no effect on the application
+  at all, but the system logs will now indicate which syscalls may be blocked.
+  This requires at least a Kernel version 4.14 as well as a recent [runc][runc]
+  release. Monitor the application hosts audit logs (defaults to
+  `/var/log/audit/audit.log`) or syslog entries (defaults to `/var/log/syslog`)
+  for syscalls via `type=SECCOMP` (for audit) or `type=1326` (for syslog).
+  Compare the syscall ID with those [listed in the Linux Kernel
+  sources][syscalls] and add them to the custom profile.
+
+- _Optional_: Use cluster additions like the [Security Profiles Operator][spo]
+  for profiling the application via its [log enrichment][logs] capabilities or
+  recording a profile by using its [recording feature][rec]. This makes the
+  above mentioned manual log investigation obsolete.
+
+[syscalls]: https://github.com/torvalds/linux/blob/7bb7f2a/arch/x86/entry/syscalls/syscall_64.tbl
+[spo]: https://github.com/kubernetes-sigs/security-profiles-operator
+[logs]: https://github.com/kubernetes-sigs/security-profiles-operator/blob/c90ef3a/installation-usage.md#record-profiles-from-workloads-with-profilerecordings
+[rec]: https://github.com/kubernetes-sigs/security-profiles-operator/blob/c90ef3a/installation-usage.md#using-the-log-enricher
+[strace]: https://man7.org/linux/man-pages/man1/strace.1.html
+
+#### Deploying the modified application
+
+Based on the outcome of the application tests, it may be required to change the
+application deployment by either specifying `Unconfined` or a custom seccomp
+profile. This is not the case if the application works as intended with
+`RuntimeDefault`.
+
+#### Enable the kubelet configuration
+
+If everything went well, then the feature is ready to be enabled by the kubelet
+configuration or its corresponding CLI flag. This should be done on a per-node
+basis to reduce the overall risk of missing a syscall during the investigations
+when running the application tests. If it's possible to monitor audit logs
+within the cluster, then it's recommended to do this for eventually missed
+seccomp events. If the application works as intended then the feature can be
+enabled for further nodes within the cluster.
+
+## Conclusion
```

Comment:
Are there any plans to default the flag `--secomp-default=true` in the future? I think it would be worth saying so either way in the blog post.

Reply:
We avoid “statements about the future” in the docs. In blog posts, it's fine to talk about these as intentions (or, if SIG Architecture agrees / would agree, as commitments).

Reply:
Added a sentence to mention that it may be enabled per default in one of the future versions of Kubernetes.

Reply:
This new sentence seems in conflict with this (very accurate) statement earlier in the post:
> With the graduation of seccomp, nothing has changed from an overall security perspective, because `Unconfined` is still the default. This is totally fine if you consider this from the upgrade path and backwards compatibility perspective of Kubernetes releases.

And in fact this was the reasoning behind changing the way the feature worked in https://github.com/kubernetes/enhancements/pull/2773.

Raising the prospect of enabling this by default, without any new details on how cluster admins will be able to roll it out safely at scale, seems like an empty promise at best and a vague cause for alarm at worst.

Reply:
@mmdriley yeah I get your point.

Given that the time till the blog release is only 2 days away I decided to revert the sentence and keep it as is.

Reply:
@saschagrunert thank you! I think it's a good change and the post is clearer this way.